packages/blade/src/components/Typography/BaseText/getBaseTextStyles.ts

```diff
@@ -0,0 +1,25 @@
+import makeTypography from '../../../utils/makeTypography';
+import type { BaseTextProps } from './BaseText.d';
+
+const getBaseTextStyles = ({
+  theme,
+  color,
+  fontFamily,
+  fontSize,
+  fontWeight,
+  fontStyle,
+  textDecorationLine,
+  lineHeight,
+  textAlign,
+}: BaseTextProps): string => `
+  color: ${color};
+  font-family: ${theme.typography.fonts.family[fontFamily]};
+  font-size: ${makeTypography(theme.typography.fonts.size[fontSize])};
```

Comment:
Not sure how you feel but I think it would be a lot more cleaner to use object here. 

```ts
  return {
    color,
    fontFamily: fonts.family[fontFamily],
    fontSize: makeTypography(fonts.size[fontSize]),
    fontWeight: fonts.weight[fontWeight],
    fontStyle,
    textDecorationLine,
    lineHeight: makeTypography(theme.typography.lineHeights[lineHeight]),
    ...(textAlign ? { textAlign } : {}),
  };
```

Reply:
Styled component does support this syntax nah? 🤔  https://styled-components.com/docs/advanced#style-objects

Reply:
tried this but then again TS kicks in since now styled components default types need to be figured out as it'll be object and again web and native things might have diff things for type definitions then again I'll have to split this. however, I'll try to do this in later components maybe. right now just little bit scared with a level of abstraction we are doing